Python/Product/Analysis/Intellisense/AnalysisQueue.cs

```diff
@@ -97,19 +98,20 @@ sealed class WaitAnalyzableTask : IAnalyzable {
         public event EventHandler<UnhandledExceptionEventArgs> UnhandledException;
 
         public void Enqueue(IAnalyzable item, AnalysisPriority priority) {
-            int iPri = (int)priority;
+            ThrowIfDisposed();
```

Comment:
This is a good example of what I'm trying to explain. `Enqueue` is called from `OnDocumentChangeProcessingComplete`, which is called from a thread-pool scheduled task: https://github.com/Microsoft/PTVS/blob/8470db3f81016111613d569f117c331a2f11ee6a/Python/Product/Analysis/LanguageServer/Server.cs#L556.
If `OCE` is thrown, `DoNotWait` will perfectly handle it. If `ODE` is thrown, we will get a Watson. And adding something like `IgnoreObjectDisposedException` isn't a solution, because we need to crash on `ObjectDisposedException` from other disposed objects.

Reply:
We should get Watson then. I tested with VSC, we get 'shutdown' and then process is insta-terminated. No exit comes. So it is unlikely process will be able to do much before it is killed.

Reply:
Then why do we care about calling `ThrowIfDisposed` at all? There is no value in this code.

Reply:
Might come useful for VS when it switches to actual LS from the current OOP analyzer.

Reply:
We should definitely handle this where it comes up and Watsons (ideally before we ship). But explicitly checking `ODE` is important, whereas in many cases quietly stopping work on `OCE` is fine because there's no need to abandon the overall process.

Semantically, `ODE` is the right choice here. But we will need to do some work to make it safe. Watch the PTVS test results after merging to see if a lot of things start failing as a result.

Reply:
Probably the best choice is to catch the `ODE` and throw an `OCE` with the `ODE` as the inner exception. That properly captures the cause of the cancellation in a way that simply throwing the cancellation exception does not.